src/Tools/ExternalAccess/Razor/Microsoft.CodeAnalysis.ExternalAccess.Razor.csproj

```diff
@@ -0,0 +1,38 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.CodeAnalysis.ExternalAccess.Razor</RootNamespace>
```

Comment:
I thought Razor already had an assembly for this, or did we change that when we did Find All References?

Reply:
Razor and XAML have a couple of assemblies currently, but it's not clear how they fit into the full picture. Either way:

1. The creation of the Razor assembly here provides a migration path to the new pattern
2. The FAR work added a bunch of IVTs to other layers which broke the intended encapsulation (if any)

Reply:
@rynowak I can migrate your bits to new dll for you once this checked in. we can put these all together on my Razor insertion PR.

IVT to for FAR is a strategic one. unlike other teams, they are using only interfaces we agreed on from the dlls which we planing to make public at the end of Razor work.

Reply:
I like the direction of this. 

We currently have IVT to the workspaces layer for the FAR/IDynamicFileInfo stuff. The long term plan is to make this public when stable, we plan to use this everywhere (VS4Mac, VS4Code)

We also have IVT to some VS-only features like the remote workspace. AFAIK this is VS-only forever, but you might have other ideas.

I'll be back on the office on the 3rd if we need to talk more then.